### PR TITLE
[fix-10729]The request's url given by the test case looks different f…

### LINKED ISSUE #10729: [Bug] The request's url given by the test case looks different from the controller layer
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image
image

### What you expected to happen

The request's url given by the test case which become right

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuchanghai**
…rom the controller layer. Some usages and parameter format errors

## Purpose of the pull request
fix #10729  only in class DataSourceControllerTest

## Brief change log

The request's url given by the test case looks different from the controller layer. Some usages and parameter format errors in class DataSourceControllerTest
## Verify this pull request

This pull request is already covered by existing tests, which annotated by  org.junit.Ignore  in class DataSourceControllerTest

### CODE REVIEW

**Comment by SbloodyS** [comment]
> Kudos, SonarCloud Quality Gate passed!    [Quality Gate passed](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=10734)
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=10734&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10734&metric=coverage&view=list) No Coverage information [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10734&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=10734&metric=new_duplicated_lines_density&view=list)

Please pay attention to code smell.

**Comment by fuchanghai** [comment]
Now there is no code smell in my local area, and it has been pushed up. It seems that there are still three excepted. Is there anything else I should pay attention to? @SbloodyS

**Comment by SbloodyS** [comment]
> Now there is no code smell in my local area, and it has been pushed up. It seems that there are still three excepted. Is there anything else I should pay attention to? @SbloodyS

Since you are a first-time-contributor. You do not have permission to run CI. You can ping me when you want to run CI.

**Comment by fuchanghai** [comment]
## When I decorate a class or method with public, or have a code smell
image

## when i remove public,CI will error
@SbloodyS

**Comment by SbloodyS** [comment]
> ## When I decorate a class or method with public, or have a code smell
> image
> 
> ## when i remove public,CI will error
> @SbloodyS

The reason is that you used the public method in class.

**Comment by fuchanghai** [comment]
Excuse me, please help me to run  CI

**Comment by fuchanghai** [comment]
@SbloodyS

**Comment by SbloodyS** [comment]
> Excuse me, please help me to run CI

I had run CI.

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/DataSourceControllerTest.java [hunk 1]
```diff
@@ -19,19 +19,23 @@
...
 import org.apache.dolphinscheduler.api.utils.Result;
 import org.apache.dolphinscheduler.common.utils.JSONUtils;
-import org.apache.dolphinscheduler.plugin.datasource.mysql.param.MySQLDataSourceParamDTO;
 
 import java.util.HashMap;
 
-import org.junit.Assert;
-import org.junit.Ignore;
-import org.junit.Test;
+import org.junit.*;
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/DataSourceControllerTest.java]: Please avoid import *. This violates checkstyle.

**Comment by fuchanghai** [comment]
@SbloodyS i had remove import *.thanks for your patience and help.please run CI again

**Comment by SbloodyS** [comment]
> @SbloodyS i had remove import *.thanks for your patience and help.please run CI again

I had approved to run CI.

**Comment by fuchanghai** [comment]
@SbloodyS If you have time, can you help me take a look. This error does not seem to be a code error, but the test case does not exist in the database, or the database information cannot be connected

**Comment by SbloodyS** [comment]
> @SbloodyS If you have time, can you help me take a look. This error does not seem to be a code error, but the test case does not exist in the database, or the database information cannot be connected

You are missing the logic of MySQL database initialization. It is recommended that you debug and execute unit tests locally.

**Comment by fuchanghai** [comment]
> > @SbloodyS If you have time, can you help me take a look. This error does not seem to be a code error, but the test case does not exist in the database, or the database information cannot be connected
> 
> You are missing the logic of MySQL database initialization. It is recommended that you debug and execute unit tests locally.

i'm so sorry for that. sonar recommended that remove 'public' modifier from method and class. Otherwise, it will prompt code smells.After I removed it, I didn't test it, which resulted in an error when I removed it.(initialzationError  which beacuse of remove 'public' modifier ). i had tested in my local. 
there only two mistake that not seem to be a code error ( can't find resource by datasouceId  and can't connect database) please run CI

**Comment by fuchanghai** [comment]
image
it really work in my local 。there mistake which  is can't find resouce by id. should i add @Ignore?

**Comment by fuchanghai** [comment]
image
@ParameterizedTest 不能和@Test 一起使用。if idon't use @ParameterizedTest on some method. will have code smells.i don't know how to balance them。can you give me some adivce?

**Comment by SbloodyS** [comment]
It seems the error the reason for the error is that the checkstyle check failed. You can use `https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html` to check it locally.

**Comment by fuchanghai** [comment]
@SbloodyS

**Comment by fuchanghai** [comment]
@SbloodyS i think i should not replace these 3 tests with a single Parameterized one.because here are three interfaces with different functions

**Comment by SbloodyS** [comment]
> @SbloodyS i think i should not replace these 3 tests with a single Parameterized one.because here are three interfaces with different functions

Okey. That's makes sense to me.

**Review by SbloodyS — APPROVED** [review]
LGTM

**Comment by fuchanghai** [comment]
@caishunfeng please review

**Comment by caishunfeng** [comment]
Thanks for your contribution @fuchanghai :smile:

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/DataSourceControllerTest.java [hunk 2]
```diff
@@ -17,21 +17,26 @@
...
 import org.apache.dolphinscheduler.api.utils.Result;
 import org.apache.dolphinscheduler.common.utils.JSONUtils;
-import org.apache.dolphinscheduler.plugin.datasource.mysql.param.MySQLDataSourceParamDTO;
 
 import java.util.HashMap;
 
 import org.junit.Assert;
 import org.junit.Ignore;
 import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
```
Comment (EricGao888) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/DataSourceControllerTest.java:36]: @fuchanghai We are not supposed to have both jUnit5 and jUnit4 style UTs in one file.

**Pull request merged**